internal/util/flowgraph/input_node.go

```diff
@@ -140,6 +152,33 @@ func (inNode *InputNode) Operate(in []Msg) []Msg {
 		msg.SetTraceCtx(ctx)
 	}
 
+	// skip timetick message feature
+	if inNode.role == typeutil.DataNodeRole &&
+		len(msgPack.Msgs) > 0 &&
+		paramtable.Get().DataNodeCfg.FlowGraphSkipModeEnable.GetAsBool() {
+		if msgPack.Msgs[0].Type() == commonpb.MsgType_TimeTick {
+			if inNode.skipMode {
+				// if empty timetick message and in skipMode, will skip some of the timetick messages to reduce downstream work
+				if inNode.skipCount == paramtable.Get().DataNodeCfg.FlowGraphSkipModeSkipNum.GetAsInt() {
+					inNode.skipCount = 0
+				} else {
+					inNode.skipCount = inNode.skipCount + 1
+					return []Msg{}
+				}
+			} else {
+				cd := paramtable.Get().DataNodeCfg.FlowGraphSkipModeColdTime.GetAsInt()
+				if time.Since(inNode.lastNotTimetickTime) > time.Second*time.Duration(cd) {
```

Comment:
```suggestion
				cd := paramtable.Get().DataNodeCfg.FlowGraphSkipModeColdTime.GetAsDuration(time.Second)
				if time.Since(inNode.lastNotTimetickTime) > cd {
```